### ISSUE TITLE
Ruff runs all pydocstyle rules when the convention is specified in the config toml files

### ISSUE THREAD

**Issue by felix-cw** [charliermarsh/ruff#3684]
This behaviour is new in 0.0.258

When running ruff with the `--select` option in a folder with a `pyproject.toml` or `ruff.toml` which specifies the pydocstyle convention,
ruff always runs all the pydocstyle rules.

Example:

```shell
# I want to check for unsorted imports
ruff check ruff_doc.py --select=I
```

`ruff_doc.py`

```python
"""Useful mathematical functions."""


def add_numbers(number1, number2):
    """Add 2 numbers.

    Args:
        number1 (Number): first number
        number2 (Number): second number

    Returns:
        Number: result of addition
    """
    return number1 + number2

```

`pyproject.toml`

```toml
[tool.ruff.pydocstyle]
convention = "google"
```

The result is 

```
ruff_doc.py:5:5: D213 [*] Multi-line docstring summary should start at the second line
ruff_doc.py:5:5: D407 [*] Missing dashed underline after section ("Args")
ruff_doc.py:5:5: D407 [*] Missing dashed underline after section ("Returns")
```

I do not expect any output, as I didn't select `D` in my command, and these error codes should be turned off in the google convention.

This behaviour is particularly bad in e.g. VSCode with the organize imports functionality, as it also tries to fix all docstring issues in all the conventions, which results in more errors.

**Comment by JonathanPlasse** [comment]
`convention` enables the `D` rules corresponding to it implicitly. You should not set `convention`if you do not want to check the `D` rules.

For vscode, you should check you are not using fix all in your configuration.

**Comment by felix-cw** [comment]
I am getting errors for rules D213 and D407, which are not included in the google convention per http://www.pydocstyle.org/en/stable/error_codes.html#default-conventions

Furthermore I am using `--select=I` but the 'D' rules are getting enabled as well. If I set `convention`, does that mean there is no way to prevent 'D' rules from being checked?

**Comment by charliermarsh** [comment]
Yeah I've confirmed that it's a regression. I'll ship a fix today.
